src/guidellm/preprocess/dataset.py

```diff
@@ -0,0 +1,224 @@
+        dataset.to_parquet(str(output_path))
+    else:
+        raise ValueError(
+            f"Unsupported file suffix '{output_path.suffix}' in output_path "
+            f"'{output_path}'. Only {SUPPORTED_TYPES} are supported."
+        )
+
+
+def _validate_output_suffix(output_path: Union[str, Path]) -> None:
+    output_path = Path(output_path)
+    suffix = output_path.suffix.lower()
+    if suffix not in SUPPORTED_TYPES:
+        raise ValueError(
+            f"Unsupported file suffix '{suffix}' in output_path '{output_path}'. "
+            f"Only {SUPPORTED_TYPES} are supported."
+        )
+
+
+def process_dataset(
+    input_data: Union[str, Path],
+    output_path: Union[str, Path],
+    processor: Union[str, Path, PreTrainedTokenizerBase],
+    processor_args: Optional[dict[str, Any]] = None,
+    data_args: Optional[dict[str, Any]] = None,
+    short_prompt_strategy: ShortPromptStrategy = ShortPromptStrategy.IGNORE,
+    pad_token: Optional[str] = None,
+    prompt_tokens_average: int = 10,
+    prompt_tokens_stdev: Optional[int] = None,
+    prompt_tokens_min: Optional[int] = None,
+    prompt_tokens_max: Optional[int] = None,
+    prompt_random_seed: int = 42,
+    output_tokens_average: int = 10,
+    output_tokens_stdev: Optional[int] = None,
+    output_tokens_min: Optional[int] = None,
+    output_tokens_max: Optional[int] = None,
+    output_random_seed: int = 123,
+    push_to_hub: bool = False,
+    hub_dataset_id: Optional[str] = None,
+) -> None:
+    _validate_output_suffix(output_path)
+    logger.info(
+        f"Starting dataset conversion | Input: {input_data} | "
+        f"Output directory: {output_path}"
+    )
+
+    dataset, column_mappings = guidellm_load_dataset(
+        input_data, data_args, processor, processor_args
+    )
+    tokenizer = check_load_processor(
+        processor,
+        processor_args,
+        "dataset conversion.",
+    )
+    prompt_column = column_mappings.get("prompt_column")
+    output_column = column_mappings.get(
+        "output_tokens_count_column", "output_tokens_count"
+    )
+
+    prompt_token_sampler = iter(
+        IntegerRangeSampler(
+            average=prompt_tokens_average,
+            variance=prompt_tokens_stdev,
+            min_value=prompt_tokens_min,
+            max_value=prompt_tokens_max,
+            random_seed=prompt_random_seed,
+        )
+    )
+
+    output_token_sampler = iter(
+        IntegerRangeSampler(
+            average=output_tokens_average,
+            variance=output_tokens_stdev,
+            min_value=output_tokens_min,
+            max_value=output_tokens_max,
+            random_seed=output_random_seed,
+        )
+    )
+
+    dataset_iterator = iter(dataset)
+    processed_prompts = []
+    handler = STRATEGY_HANDLERS[short_prompt_strategy]
+
+    for prompt_row in dataset_iterator:
+        prompt_text = prompt_row[prompt_column]
+        target_prompt_len = next(prompt_token_sampler)
```

Comment:
Mentioned this above in the handlers, but I think we end up with issues here for sampling to create the target_prompt_len and treating it as the min since we could have lengths that exceed our maximum settings and additionally we might throw away or combine data that technically would have worked within the target ranges the user set

Reply:
This gets tricky, though, towards accepting a value within the stats range since that could shift our stats distribution